### PR TITLE
compiler: fix minor issues detected by JET

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

**Comment by aviatesk** [comment]
@nanosoldier `runbenchmarks("inference", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/47163#issuecomment-1278680620) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/0d0fe2a_vs_f424431/report.md).

File: base/compiler/tfuncs.jl [hunk 1]
```diff
@@ -2393,7 +2393,7 @@ function setglobal!_nothrow(argtypes::Vector{Any})
     end
     M, s, newty = argtypes
     if M isa Const && s isa Const
-        M, s = M.val, s.val
+        M, s = M.val::Module, s.val::Symbol
```
Comment (vtjnash) [inline base/compiler/tfuncs.jl:2396]: do we know this is true, or do we need to check it?

Comment (aviatesk) [reply base/compiler/tfuncs.jl:2396]: I thought we have some prior check, but it doesn't seem to exist. We need type checks here.

**Review by vtjnash — APPROVED** [review]

**Pull request merged**